File: Back-End/routes/videoCallData.js
const express = require("express");
const router = express.Router();
const db = require("../utils/DB");

router.get('/videoCall/:appointmentId', async (req, res) => {
  const appointmentId = req.params.appointmentId;

  try {
    const [rows] = await db.query(
      'SELECT channel_name, doctor_uid, patient_uid FROM appointments WHERE appointment_id = ?',
      [appointmentId]
    );

    if (rows.length === 0) {
      return res.status(404).json({ error: 'Appointment not found' });
    }

    res.json(rows[0]);
  } catch (error) {
    console.error('Error fetching video call info:', error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

module.exports = router;

File: Front-End/lib/models/videoCallData.dart
class VideoCallData {
  final String channelName;
  final int doctorUid;
  final int patientUid;

  VideoCallData({
    required this.channelName,
    required this.doctorUid,
    required this.patientUid,
  });

  factory VideoCallData.fromJson(Map<String, dynamic> json) {
    return VideoCallData(
      channelName: json['channel_name'],
      doctorUid: json['doctor_uid'],
      patientUid: json['patient_uid'],
    );
  }
}
